Makefile

```diff
@@ -86,6 +97,160 @@ endif
 	@docker push $(DOCKER_REPO):$(VERSION)
 	@echo "==> Your image is now available at $(DOCKER_REPO):$(VERSION)"
 
+# runner-build builds the e2e test runner image. Sadly, this is much harder to
+# do cache-efficiently than it should be for Docker multi-stage builds since
+# those require build targets to be pushed and referenced individually. See
+# https://andrewlock.net/caching-docker-layers-on-serverless-build-hosts-with-multi-stage-builds---target,-and---cache-from/
```

Comment:
😞 that this needs to be done but 👏 amazing job discovering a way to utilize cache to its max

Reply:
Looking for the German word that expresses feeling both pride and rejection about one's work. :)